File: ReinforcementLearning/domain/Aircraft.py

class Aircraft:
    def __init__(self, id, type, crewNum, flightCost, layoverCost, quickTurnCost):
        self.id = id
        self._type = type
        self._crewNum = crewNum
        self._flightCost = flightCost
        self._layoverCost = layoverCost
        self._quickTurnCost = quickTurnCost
        # self.name = "Aircraft - " + type # 인스턴스 이름

    def find_aircraft(self, aircrafts, name):
        aircraft = next(
            (temp for temp in aircrafts if temp['type'] == name), None)
        if aircraft is None:
            raise ValueError("Aircraft not found")
        return aircraft

    @property
    def type(self):
        return self._type

    @type.setter
    def type(self, value):
        self._type = value

    @property
    def crewNum(self):
        return self._crewNum

    @crewNum.setter
    def crewNum(self, value):
        self._crewNum = value

    @property
    def flightCost(self):
        return self._flightCost

    @flightCost.setter
    def flightCost(self, value):
        self._flightCost = value

    @property
    def layoverCost(self):
        return self._layoverCost

    @layoverCost.setter
    def layoverCost(self, value):
        self._layoverCost = value

    @property
    def quickTurnCost(self):
        return self._quickTurnCost

    @quickTurnCost.setter
    def quickTurnCost(self, value):
        self._quickTurnCost = value

    def __str__(self):
        return f"Aircraft - {self.type}"

File: ManagerProgram/src/main/java/org/dongguk/domain/Aircraft.java
package org.dongguk.domain;

import lombok.*;
import org.dongguk.AbstractPersistable;

import java.util.List;

@Getter
@Setter
public class Aircraft extends AbstractPersistable {
    // 기종 이름
    final private String type;
    // 기종 크루원 수
    final private int crewNum;
    // 비행 수당
    final private int flightCost;
    // Layover 비용
    final private int layoverCost;
    // QuickTurn 비용
    final private int quickTurnCost;

    @Override
    public String toString() {
        return "Aircraft - " + type;
    }

    public static Aircraft of(final List<Aircraft> aircrafts, final String name) {
        return aircrafts.stream()
                .filter(temp -> temp.getType().equals(name))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Aircraft not found"));
    }

    @Builder
    public Aircraft(long id, String type, int crewNum, int flightCost, int layoverCost, int quickTurnCost) {
        this.id = id;
        this.type = type;
        this.crewNum = crewNum;
        this.flightCost = flightCost;
        this.layoverCost = layoverCost;
        this.quickTurnCost = quickTurnCost;
    }
}
